### PR TITLE
 project setup

### LINKED ISSUE #4: Project Setup
[Project Setup]

### Acceptance Criteria
- File Structure
- Server Setup
- Client (React) Setup
- ESLINT config

---

### File Strucutre #4 
  - Client
  - Server
  - Testing
   #5
### Server Setup
  - Install  modules :
     - express
     - jsonwebtoken
     - yup
     - dotenv
     - cookie-parser
     - compression
     - pg
    - nodemon
    - supertest
    - jest
    - eslint
    - pre-commit

Setup the server and make it listen on port `5000`

### React Setup
Generate react app for the client using command 
```
npx create-react-app 
```
  - Install  modules :
     -  prop-types
     -  react-router-dom
     -  axios
     -  material-ui

- add Proxy to client package.json
```json
"proxy": "http://localhost:5000/"
```
### ESLINT config
  - ESLINT for the Client
    ```javascript
        eslint
        eslint-config-airbnb-base
        eslint-plugin-import
        eslint-config-prettier
        eslint-plugin-prettier
        eslint-plugin-flowtype
        eslint-plugin-jsx-a11y
        eslint-plugin-react
        eslint-plugin-react-hooks
        prettier
      ```

  - ESLINT for the Server
    ```javascript
        eslint
        eslint-config-airbnb
        eslint-plugin-prettier
        save-exact prettier
        eslint-config-prettier
        prettier
        eslint-plugin-jest

      ```
---

### PR DESCRIPTION
**Pull request by IsraaHamdi**
project setup
close #4 
Relates #4

### CODE REVIEW

**Review by Jehad91 — DISMISSED** [review]

**Review by MostafaQanoo — DISMISSED** [review]

**Review by MohammadAlHabil — DISMISSED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "module": "commonjs",
+    // "isolatedModules": false,
+    "esModuleInterop": true,
+    "outDir": "dist",
+    "rootDir": "./", 
+    "moduleResolution": "node",
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true 
+  }
```
Comment (Mu7ammadAbed) [inline tsconfig.json:13]:
```suggestion
  },
  "include": ["server/**/*.ts"],
```

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "target": "es6",
+    "module": "commonjs",
+    // "isolatedModules": false,
+    "esModuleInterop": true,
+    "outDir": "dist",
+    "rootDir": "./",
```
Comment (Mu7ammadAbed) [inline tsconfig.json:8]:
```suggestion
```

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "target": "es6",
+    "module": "commonjs",
+    // "isolatedModules": false,
```
Comment (Mu7ammadAbed) [inline tsconfig.json:5]:
```suggestion
    "isolatedModules": false,
```

File: server/app.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import express, { Request, Response, Application } from 'express';
+import dotenv from 'dotenv';
+import cookieParser from 'cookie-parser';
+import compression from 'compression';
+
+const app:Application = express();
+dotenv.config();
+app.set('port', process.env.PORT || 5000);
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
```
Comment (Mu7ammadAbed) [inline server/app.ts:11]:
```suggestion
app.use(express.urlencoded({ extended: false}));
```

File: server/app.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+app.use(cookieParser());
+app.use(compression());
+
+app.get('/', (req:Request, res:Response):void => {
+  res.json({ msg: 'Hello CleanStead' });
+});
```
Comment (Mu7ammadAbed) [inline server/app.ts:17]: We should serve this route at the development only, and you should change the response body template from `msg` to `message`
```suggestion
if (NODE_ENV === 'development') {
  app.get('/', (req:Request, res:Response):void => {
    res.json({ message: 'Hello CleanStead' });
  });
}
```

File: server/app.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { join } from 'path';
+import express, { Request, Response, Application } from 'express';
+import dotenv from 'dotenv';
+import cookieParser from 'cookie-parser';
+import compression from 'compression';
+
+const app:Application = express();
+dotenv.config();
+app.set('port', process.env.PORT || 5000);
```
Comment (Mu7ammadAbed) [inline server/app.ts:9]:
```suggestion
dotenv.config();
const {
  env: { PORT, NODE_ENV },
} = process;

app.set('port', PORT || 5000);
app.disable('x-powered-by');
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by Mu7ammadAbed — COMMENTED** [review]
This is a very great job, there are only a few notes I mentioned above, for any explanation you can reach me at any time.

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Jehad91 — APPROVED** [review]

**Review by MostafaQanoo — APPROVED** [review]

**Review by MohammadAlHabil — APPROVED** [review]

**Pull request merged**